### PR TITLE
feat(client): Alipay social plugin

### PR DESCRIPTION
**Pull request by gao-sun**
## Summary

implement Alipay social plugin

note we are using a separate repo https://github.com/logto-io/social-sdks for maintaining "edited official SDKs" (Alipay doesn't have an official xcframework) to leverage [SPM binary dependency](https://github.com/apple/swift-evolution/blob/main/proposals/0272-swiftpm-binary-dependencies.md). 

to-do: LOG-2084 to use global scheme for 3rd party app to go back

## Linear Issue Reference

LOG-2038

## Testing

https://user-images.githubusercontent.com/14722250/161216194-7ea2aebf-0a63-4729-b277-8d280a9f7311.MP4

and it works!

image

## Reviewers

@logto-io/eng

### CODE REVIEW

File: Demos/SwiftUI Demo/SwiftUI Demo/ContentView.swift [hunk 1]
```diff
@@ -121,6 +123,19 @@ struct ContentView: View {
                     }
                 }
             }
+
+            Button("Alipay") {
+                alipayPlugin.start(LogtoSocialPluginConfiguration(
+                    redirectUri: URL(string: "alipay://?app_id=some_app_id&state=foo")!,
```
Comment (demonzoo) [inline Demos/SwiftUI Demo/SwiftUI Demo/ContentView.swift:129]: How does it work with an app id this arbitrary?

Comment (gao-sun) [reply Demos/SwiftUI Demo/SwiftUI Demo/ContentView.swift:129]: replaced before commit :-)

**Review by demonzoo — APPROVED** [review]
Overal LGTM. But honestly don't quite understand some of the syntax usages in Swift...

File: Demos/SwiftUI Demo/SwiftUI Demo/ContentView.swift (hunk 1 above)
Comment (demonzoo) [reply Demos/SwiftUI Demo/SwiftUI Demo/ContentView.swift:129]: Aha.. no wonder

**Comment by gao-sun** [comment]
@demonzoo yes it's a totally different language

**Comment by gao-sun** [comment]
@demonzoo i've updated to another version, please check description and code again

File: Sources/LogtoClient/LogtoClient/LogtoClient.swift [hunk 2]
```diff
@@ -53,6 +68,31 @@ public class LogtoClient {
...
+    // MARK: Internal Functions
+
+    func handle(notification: Notification) {
+        guard let object = notification.object as? NotificationObject else {
+            return
+        }
+
+        guard object.clientId == nil || object.clientId == logtoConfig.clientId else {
+            return
+        }
```
Comment (simeng-li) [inline Sources/LogtoClient/LogtoClient/LogtoClient.swift:91]: nit: per discussion could add some comments here.

File: Sources/LogtoSocialPluginAlipay/LogtoSocialPluginAlipay.swift [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        components.queryItems = [
+            URLQueryItem(name: "auth_type", value: "PURE_OAUTH_SDK"),
+            URLQueryItem(name: "scope", value: "auth_user"),
+        ] + (redirectComponents.queryItems ?? [])
+
+        AFServiceCenter.call(.auth, withParams: [
+            kAFServiceOptionBizParams: [
+                "url": components.string,
+            ],
+            kAFServiceOptionCallbackScheme: "logto-demo",
```
Comment (simeng-li) [inline Sources/LogtoSocialPluginAlipay/LogtoSocialPluginAlipay.swift:63]: not sure what is this for ,but wondering for production use, where should this come from?

**Review by wangsijie — APPROVED** [review]

File: Sources/LogtoSocialPluginAlipay/LogtoSocialPluginAlipay.swift (hunk 3 above)
Comment (gao-sun) [reply Sources/LogtoSocialPluginAlipay/LogtoSocialPluginAlipay.swift:63]: per PR description: to-do: [LOG-2084](https://linear.app/silverhand/issue/LOG-2084) to use global scheme for 3rd party app to go back

**Pull request merged**